### PR TITLE
Sema: report inconsistent access levels on imports of one module from one file

### PR DESCRIPTION
**Pull request by xymus**
Warn on imports with different access levels of the same target module from the same source file. This situation can lead to unexpected behavior as the compiler will only take into account the most permissive import.

In the following example, the module Foundation in functionally imported as public. The `internal` modifier will be ignored. The new diagnostic reports it as such.

```swift
public import Foundation
internal import Foundation // warning: module 'Foundation' is imported as 'public' from the same file; this 'internal' access level will be ignored
```

This could be an easy mistake to do when using scoped imports where this diagnostic will also help.

rdar://117855481

### CODE REVIEW

**Comment by xymus** [comment]
@swift-ci Please smoke test

File: test/Sema/access-level-import-inconsistent-same-file.swift [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+// RUN:   -swift-version 5
+
+//--- Lib.swift
+public struct Type1 {}
+public struct Type2 {}
+
+//--- Client.swift
+/// Simple public vs internal.
+public import Lib1 // expected-note {{imported 'public' here}}
+// expected-warning @-1 {{public import of 'Lib1' was not used in public declarations or inlinable code}}
```
Comment (tshortli) [inline test/Sema/access-level-import-inconsistent-same-file.swift:29]: If this test is not specifically meaning to exercise the `was not used in public declarations` diagnostic, I would find it easier to follow if this diagnostic were suppressed by some trivial use of declarations from the library.

**Review by tshortli — APPROVED** [review]

**Review by artemcm — APPROVED** [review]

File: test/Sema/access-level-import-inconsistent-same-file.swift (hunk 1 above)
Comment (xymus) [reply test/Sema/access-level-import-inconsistent-same-file.swift]: I followed your suggestion and all of these messages are gone, except one which is a bug I'll have to fix in the other diagnostic logic.

**Comment by xymus** [comment]
@swift-ci Please smoke test

**Pull request merged**